core/src/main/java/com/godaddy/commerce/core/LoggingHelper.java

```diff
@@ -19,11 +21,23 @@ public static LogstashMarker fromRoutingContext(RoutingContext routingContext, S
     addIfNonNull(marker, "customerId", getRequestParam(routingContext, "customerId"));
     addIfNonNull(marker, "ownerId", getRequestParam(routingContext, "ownerId"));
     addIfNonNull(marker, "route", getRoute(routingContext));
-    marker.add(withRequestId(routingContext));
+    Marker requestIdMarker = withRequestId(routingContext);
+    if (requestIdMarker != null) {
+      marker.add(requestIdMarker);
+    }
     return marker;
   }
 
   public static Marker withRequestId(RoutingContext routingContext) {
+    if (Context.isOnEventLoopThread()) {
```

Comment:
> [<img alt="ukurylenka" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.secureserver.net/ukurylenka) **Authored by [ukurylenka](https://github.secureserver.net/ukurylenka)**
_<time datetime="2021-12-23T12:52:30Z" title="Thursday, December 23rd 2021, 1:52:30 pm +01:00">Dec 23, 2021</time>_

---


So the only difference between event loop and worker threads is that for event loop we don't check MDC?
If so combining checks should help avoid copy and paste code (lines 33-35 looks identically to 41-43)
```suggestion
if (Context.isOnEventLoopThread() || StringUtils.isBlank(MDC.get(XRequestIdUtility.X_REQUEST_ID_LOGGING)){
      String requestIdVal =
          StringUtils.defaultIfBlank(XRequestIdUtility.get(routingContext), "empty");
      return Markers.append(XRequestIdUtility.X_REQUEST_ID_LOGGING, requestIdVal);
    }
return null;
```

Reply:
> [<img alt="emoiseev" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.secureserver.net/emoiseev) **Authored by [emoiseev](https://github.secureserver.net/emoiseev)**
_<time datetime="2021-12-23T12:55:01Z" title="Thursday, December 23rd 2021, 1:55:01 pm +01:00">Dec 23, 2021</time>_

---


Good catch.
Also, notice that, but was too late).
If I won't forget I'll fix that in the next ticket. Hope it's ok.